.travis.yml

```diff
@@ -83,7 +95,17 @@ script:
 # Wait for kube-dns to be ready.
 - JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl -n kube-system get pods -lk8s-app=kube-dns -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1;echo "waiting for kube-dns to be available"; kubectl get pods --all-namespaces; done
 - helm install -f helm-configs/.travis-galaxy-sc-deployment.yaml --name travis --version $HELM_REPO_VERSION galaxy-helm-repo/galaxy-stable
-- planemo lint -r --report_level error tools
+# Run planemo linting and testing for repositories changed in these commits
+- cat changed_tools.list
+- set -e
+- |
+  while read -r TOOL; do
+    planemo lint --report_level error "$TOOL"
+  done < changed_tools.list
+- |
+  while read -r TOOL; do
+    planemo test --galaxy_python_version 3.7 "$TOOL"
+  done < changed_tools.list
```

Comment:
I think that instead of testing per tool, we should test per directory, then it doesn't start and stop a galaxy instance every time...